### PR TITLE
add SyncBatchNorm support for training examples

### PR DESCRIPTION
**Pull request by Sharpiless**
## Checklist before submitting

- [ ] Did you read the [contributor guide](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md)?
- [ ] Did you update the docs?
- [ ] Did you write any tests to validate this change?  
- [ ] Did you update the [CHANGELOG](https://github.com/horovod/horovod/blob/master/CHANGELOG.md), if this change affects users?

## Description

Fixes # (issue).

## Review process to land 

1. All tests and other checks must succeed.
2. At least one member of the [technical steering committee](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md) must review and approve.
3. If any member of the technical steering committee requests changes, they must be addressed.

Replace BatchNorm in pytorch examples with  SyncBatchNorm.

Mutil-GPU training with BatchNorm with lead to different BN mean and var between different GPU, and the final accuracy will be lower.

### CODE REVIEW

**Review by tgaddair — COMMENTED** [review]
Thanks for the fix @Sharpiless! Can you make this optional through a command line arg like `--sync-bn`, as it may have an effect on performance?
